File: chapter04/src/main/java/com/ixiongyu/interfaces/javaversion/Clickable.java
package com.ixiongyu.interfaces.javaversion;

/**
 * @author xiongyu
 * @version Create at ：2020/8/23 下午8:44
 */
public interface Clickable {
    default void showOff() {
        System.out.println("Clickable showOff");
    }

    void click();
}

File: chapter04/src/main/java/com/ixiongyu/interfaces/Clickable.kt
package com.ixiongyu.interfaces

/**
 * @author  xiongyu
 * @version Create at ：2020/8/23 下午8:34
 */
interface Clickable {
    fun click()
    //默认方法可以直接使用，也可以在实现内中有自己的实现
    fun showOff() = println("Clickable showOff")
}
